openstack/standalone-cinder/pkg/volumeservice/actions.go

```diff
@@ -108,11 +111,27 @@ func CreateCinderVolume(vs *gophercloud.ServiceClient, options controller.Volume
 // become available.  The connection information cannot be retrieved from cinder
 // until the volume is available.
 func WaitForAvailableCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error {
-	// TODO: Implement proper polling instead of brain-dead timers
 	c := make(chan error)
-	go time.AfterFunc(5*time.Second, func() {
-		c <- nil
-	})
+	go func() error {
+		num := 0
+		for {
+			time.Sleep(sleepTime * time.Second)
+			if num > maxRetry {
+				err := errors.New("Failed to get volume status after max retries")
+				c <- err
+			}
+			volume, err := volumes_v2.Get(vs, volumeID).Extract()
```

Comment:
You might consider adding a vsbroker interface for volumes_v2.Get.   See #453.  We want to add unit tests for this method to ensure that it behaves as expected in the good case, when max retries are exceeded, and when we fail to get the volume information whatsoever.

Reply:
Agree that It's better to add interface/function "getCinderVolumeState" into vsbroker/actions,  and reuse it in function "waitForAvailableCinderVolume"(could also be reused in other cases).  But it will be overlap with what did in #453